src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -142,12 +160,25 @@ public async Task<T> GetConfigurationAsync(CancellationToken cancel)
                         _currentConfiguration = await _configRetriever.GetConfigurationAsync(MetadataAddress, _docRetriever, CancellationToken.None).ConfigureAwait(false);
```

Comment:
I am thinking not to throw is bad config is received as _currentConfiguration may be fine.
throwing may also interfere with LKG.
```cs
try
{
    var configuration = await _configRetriever.GetConfigurationAsync(MetadataAddress, _docRetriever, CancellationToken.None).ConfigureAwait(false);
    if (_configValidator != null)
    {
        ConfigurationValidationResult result = _configValidator.Validate(_currentConfiguration);
        if (result.Succeeded)
            _currentConfiguration = configuration
        else
           LogHelper.LogWarning(LogMessages.IDX20810, result.Exception);
    }
    
    _currentConfiguration = configuration;
}

Reply:
I'm assuming the example above should be:

...
   if (_configValidator != null && !_configValidator.Validate(configuration).Succeeded)
...

right (instead of checking if _currentConfiguration is invalid which is the stale one?

Seems fine to me. Would have to be sure not to update _syncAfter or make it try again soon at least right? Something like 5min+jitter?

I think continuing to send back the stale config instead of failing is ok here should the stale version be validated as well? If _configValidator.Validate() fails for _currentConfiguration && configuration seems like it would be worth throwing then, no?

Reply:
@TimHannMSFT that is what we do today, in the case we couldn't get new metadata.
In this case, i am suggesting we just skip any invalid metadata.